### ISSUE TITLE
pylint cannot find third-party packages in virtual environment

### ISSUE THREAD

**Issue by GNUqb114514** [pylint-dev/pylint#9720]
### Bug description

```python
I created a python virtual environment, used a package in the environment, then invoke pylint.
It can't find the package in the virtual environment, and give me an E0401.
```

### Command used

```shell
make lint
(it actually does 'pylint **/*.py')
```

### Pylint output

```python
************* Module fetching
src/fetching.py:5:0: E0401: Unable to import 'imapclient' (import-error)

-------------------------------------------------------------------
Your code has been rated at 9.50/10 (previous run: 10.00/10, -0.50)
```

### Expected behavior

No `E0401`

### Pylint version

```shell
pylint 2.16.2
astroid 2.14.2
Python 3.11.2 (main, Mar 13 2023, 12:18:29) [GCC 12.2.0]
```

### OS / Environment

Debian 12 bookworm

### Additional dependencies

```python
IMAPClient==3.0.1
iterfzf==1.4.0.51.0
```

**Comment by DanielNoord** [comment]
Is `pylint` also coming from the virtualenv? Or is your `make` script using `pylint` from the global environment?

**Comment by GNUqb114514** [comment]
It is coming from the package manager; and if I add the environment variable PYTHONPATH it will work fine.


-------- 原始邮件 --------
发件人： "Daniël van Noord" ***@***.***>
发送时间： 2024年6月10日 GMT+08:00 15:05:26
收件人： pylint-dev/pylint ***@***.***>
抄送： GNUqb114514 ***@***.***>, Author ***@***.***>
主题： Re: [pylint-dev/pylint] pylint cannot find third-party packages in virtual environment (Issue #9720)

Is `pylint` also coming from the virtualenv? Or is your `make` script using `pylint` from the global environment?

**Comment by DanielNoord** [comment]
Then this is expected behaviour. `pylint` can only find modules in the same environment as it is installed. If you install `pylint` using the global package manager and use a `virtualenv` for you program's dependencies they won't be found.
